pages/api/admin/sso-tracer/[traceId]/index.ts

```diff
@@ -1,37 +1,22 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
 import jackson from '@lib/jackson';
-import type { IAdminController } from '@boxyhq/saml-jackson';
+import { defaultHandler } from '@lib/api';
 
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
-  const { method } = req;
-  try {
-    const { adminController } = await jackson();
-
-    switch (method) {
-      case 'GET':
-        return await handleGET(req, res, adminController);
-      default:
-        res.setHeader('Allow', 'GET');
-        res.status(405).json({ error: { message: `Method ${method} Not Allowed` } });
-    }
-  } catch (error: any) {
-    const { message, statusCode = 500 } = error;
-
-    return res.status(statusCode).json({ error: { message } });
-  }
+  await defaultHandler(req, res, {
+    GET: handleGET,
+  });
 };
 
 // Get SAML Trace by traceId
-const handleGET = async (req: NextApiRequest, res: NextApiResponse, adminController: IAdminController) => {
+const handleGET = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { adminController } = await jackson();
+
   const { traceId } = req.query as { traceId: string };
 
   const trace = await adminController.getSSOTraceById(traceId);
 
-  if (!trace) {
-    return res.status(404).json({ error: { message: 'Trace not found.' } });
-  }
```

Comment:
How are we handling trace being undefined or null i.e. 404 error here?

Reply:
@ukrocks007 https://github.com/boxyhq/jackson/blob/main/npm/src/controller/admin.ts#L54 .The error thrown here will be caught in the defaultHandler catch.